### PR TITLE
Add GET Individual Full Record

### PR DESCRIPTION
**Pull request by PhilipGlover**
IDVA3-2394 - new endpoint for `/company/{company_number}/persons-with-significant-control/individual/{notification_id}/full_record`
IDVA3-2396 - add to service to retrieve data and return new model 
IDVA3-2397 - Internal API key user interceptor

Screenshot 2024-11-25 at 2 55 24 PM

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/pscdataapi/config/WebSecurityConfig.java [hunk 1]
```diff
@@ -12,23 +14,31 @@
...
 import uk.gov.companieshouse.api.filter.CustomCorsFilter;
 import uk.gov.companieshouse.api.interceptor.InternalUserInterceptor;
 import uk.gov.companieshouse.api.interceptor.UserAuthenticationInterceptor;
 
 @Configuration
 public class WebSecurityConfig implements WebMvcConfigurer {
+    // feature flag marked for future removal
+    @Value("${feature.psc_individual_full_record_get:false}")
+    private Boolean featurePscIndividualFullRecordGet;
+// TODO: integration test
```
Comment (kinpang-CH) [inline src/main/java/uk/gov/companieshouse/pscdataapi/config/WebSecurityConfig.java:26]: Did this integration test get added? 
if so TODO can be removed

**Review by kinpang-CH — CHANGES_REQUESTED** [review]
Address comment

File: src/main/java/uk/gov/companieshouse/pscdataapi/config/WebSecurityConfig.java (hunk 1 above)
Comment (hepsimo) [reply src/main/java/uk/gov/companieshouse/pscdataapi/config/WebSecurityConfig.java:26]: Good spot, yes this is definitely covered by `CompanyPscFullRecordGetDisabledControllerTest` and Cucumber test `get_individual_full_record.feature`
I'll remove the comment.

**Review by kinpang-CH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Pull request merged**